protocols/mdns/src/behaviour.rs

```diff
@@ -46,7 +56,11 @@ pub struct Mdns {
     if_watch: IfWatcher,
 
     /// Mdns interface states.
-    iface_states: HashMap<IpAddr, InterfaceState>,
+    #[cfg(feature = "async-io")]
+    iface_states: HashMap<IpAddr, InterfaceState<AsyncUdpSocket, AsyncTimer>>,
+
+    #[cfg(feature = "tokio")]
+    iface_states: HashMap<IpAddr, InterfaceState<TokioUdpSocket, AsyncTimer>>,
```

Comment:
Features should be additive in Rust but this code will break at compile-time if both are enabled.

I think we might need to go with an approach similar as in `libp2p-tcp` here: Create a dedicated `TokioMdns` and `AsyncIoMdns` struct and enable them conditionally.